analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -591,27 +592,61 @@ open class Npm(
-                .mapNotNull { line ->
-                    line.removePrefix(prefix).let {
-                        commonSecondaryPrefixes.fold(it) { remainder, prefix -> remainder.removePrefix(prefix) }
-                    }.takeUnless { it.isBlank() }
+            val ignorablePrefixes = setOf("code ", "errno ", "path ", "syscall ")
+            val singleLinePrefixes = setOf("deprecated ")
+            val minSecondaryPrefixLength = 5
+
+            val issueLines = lines.mapNotNull { line ->
+                line.withoutPrefix(prefix)?.takeUnless { ignorablePrefixes.any { prefix -> it.startsWith(prefix) } }
+            }
+
+            var commonPrefix: String
+            var previousPrefix = ""
+
+            val collapsedLines = issueLines.fold(mutableListOf<String>()) { messages, line ->
+                if (messages.isEmpty()) {
+                    // The first line is always added including the prefix. The prefix will be removed later.
+                    messages += line
+                } else {
+                    // Find the longest common prefix that ends with space.
+                    commonPrefix = line.commonPrefixWith(messages.last())
+                    if (!commonPrefix.endsWith(' ')) {
+                        // Deal with prefixes being used on their own as separators.
+                        commonPrefix = if ("$commonPrefix " == previousPrefix) {
+                            "$commonPrefix "
+                        } else {
+                            commonPrefix.dropLastWhile { it != ' ' }
+                        }
+                    }
+
+                    if (commonPrefix !in singleLinePrefixes && commonPrefix.length >= minSecondaryPrefixLength) {
+                        // Do not drop the whole prefix but keep the space when concatenating lines.
+                        messages[messages.size - 1] += line.drop(commonPrefix.length - 1).trimEnd()
+                        previousPrefix = commonPrefix
+                    } else {
+                        // Remove the prefix from previously added message start.
+                        messages[messages.size - 1] = messages.last().removePrefix(previousPrefix).trimStart()
+                        messages += line
+                    }
                 }
 
-            if (issueLines.isNotEmpty()) {
+                messages
+            }
+
+            if (collapsedLines.isNotEmpty()) {
+                // Remove the prefix from the last added message start.
+                collapsedLines[collapsedLines.size - 1] = collapsedLines.last().removePrefix(previousPrefix).trimStart()
+            }
+
+            collapsedLines.forEach { line ->
+                // Skip any footer as a whole.
+                if (line == "A complete log of this run can be found in:") return
+
                 issues += Issue(
                     source = managerName,
-                    message = issueLines.joinToString("\n"),
+                    message = line,
```

Comment:
Is it necessary for fixing this flakyness to split up the issue into separate ones per line?

Apart from that, would it make sense to do this split-up in a separate commit anyway?

Reply:
> Is it necessary for fixing this flakyness to split up the issue into separate ones per line?

More or less, yes. As the order of NPM `stderr` lines might be different on each run, we can either put a defined order on all `stderr` lines (independently of whether they belong to the same error / warning or not; this is the approach that was previously taken e.g. [here](https://github.com/oss-review-toolkit/ort/pull/6624/commits/ee1cff64ad6dfd134151c40664a9d4c49dd12b00#diff-8014b93031678f0763db8fee8d0f118858b435b467390a80813754ef81c83b80L68-L76)), or we can first associate the lines to their errors / warnings, and then disregard the order of the created issues (which is the new approach in this PR).

> Apart from that, would it make sense to do this split-up in a separate commit anyway?

That's a bit of introducing a change that will be changed in the next commit anyway, but I can do that for clarity.